tests/scripts/depends.py

```diff
@@ -381,7 +381,7 @@ def config_symbols_matching(self, regexp):
 
     def __init__(self, options, conf):
         """Gather data about the library and establish a list of domains to test."""
-        build_command = [options.make_command, 'CFLAGS=-Werror']
+        build_command = [options.make_command, 'CFLAGS=-Werror -O2']
```

Comment:
Why do we do this? It will increase built time...

Reply:
It makes a big difference to run time

Reply:
`-O2` increases the build time but reduces the test time. I'm not sure which one dominates. It may have changed in the history of the library and might even depend on whether the configuration includes high-CPU-consumption tests such as Brainpool curves, LMS private keys and PBKDF2.

Reply:
I guess mainly my question was about the change with no comment, but no argument against making this change

Reply:
Assuming that my local tests are representative, for user time, the build time difference for clang is about 2s (O2 vs O0), whereas test time (`make test` for default config) varies by about 18s. The build time difference will be further reduced for a parallel build, whereas tests do not run in parallel.

So the win from reduced test time is more than 9x for me, i.e. I think we should normally use -O2 if we're going to run the tests.

(Obviously in some cases we want to run the tests with different optimisation settings, but that's a small set of exceptional cases).